### PR TITLE
Make sureHtmlDiff::compareHTML is not passed null values

### PR DESCRIPTION
**Pull request by alex-dna**
## Description
When a record is saved without a title or content, when viewing the RSS changes feed, a type error occurs.

Uncaught TypeError: SilverStripe\View\Parsers\HtmlDiff::compareHtml(): Argument https://github.com/silverstripe/silverstripe-versionfeed/issues/1 ($from) must be of type array|string, null given, called in /var/www/html/vendor/silverstripe/versionfeed/src/VersionFeed.php on line 110

## Manual testing steps
I first encountered the issue with Elemental, after saving an element without a title.
But it should be the case for any page too (although it is less likely to save a page without title through the CMS).
Easiest way to reproduce is to temper with database if possible, set a page record version title or content to NULL, then visit the page URL/changes.

## Issues

- https://github.com/silverstripe/silverstripe-versionfeed/issues/95

## Pull request checklist

- [x] The target branch is correct
    - See [picking the right version](https://docs.silverstripe.org/en/contributing/code/#picking-the-right-version)
- [x] All commits are relevant to the purpose of the PR (e.g. no debug statements, unrelated refactoring, or arbitrary linting)
    - Small amounts of additional linting are usually okay, but if it makes it hard to concentrate on the relevant changes, ask for the unrelated changes to be reverted, and submitted as a separate PR.
- [x] The commit messages follow our [commit message guidelines](https://docs.silverstripe.org/en/contributing/code/#commit-messages)
- [x] The PR follows our [contribution guidelines](https://docs.silverstripe.org/en/contributing/code/)
- [x] Code changes follow our [coding conventions](https://docs.silverstripe.org/en/contributing/coding_conventions/)
- [x] This change is covered with tests (or tests aren't necessary for this change)
- [x] Any relevant User Help/Developer documentation is updated; for impactful changes, information is added to the changelog for the intended release
- [x] CI is green

### CODE REVIEW

File: src/VersionFeed.php [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
 namespace SilverStripe\VersionFeed;
 
+use SilverStripe\CMS\Model\SiteTree;
```
Comment (ChloeSartorelli) [inline src/VersionFeed.php]: Please revert the changes to the `use` statements, as they are not in any way related to the purpose of this PR

**Review by GuySartorelli — CHANGES_REQUESTED** [review]
Hi,

Thank you for your contribution.
I notice you removed the template from the PR description - I've added it back. Please fill this in, as it makes it easier for us to review your PR.

File: src/VersionFeed.php (hunk 1 above)
Comment (alex-dna) [reply src/VersionFeed.php]: Done.

**Comment by GuySartorelli** [comment]
Can you please fill in the full PR template? https://github.com/silverstripe/silverstripe-versionfeed/pull/96#pullrequestreview-1842330133

**Comment by GuySartorelli** [comment]
There are a couple of things to change which are pointed out in the checklist - in the interest of getting things moving along I'll point them out instead of asking you to go through the checklist again :p

Please make changes accordingly:

1. Please target the `3.1` branch so that this can be released as a patch (otherwise it'll need to wait until April)
    You may need to [reset your commits](https://gist.github.com/GuySartorelli/74627843aeb75e12b777b1f44cb22d2f) afterward
2. I won't block merging for this, but if you could squash your commits and use the `FIX` prefix in the commit message, that would be great.

**Comment by GuySartorelli** [comment]
@alex-dna You've pulled through some commits from the `3` branch that still need to be reset. Can you please [reset your commits](https://gist.github.com/GuySartorelli/74627843aeb75e12b777b1f44cb22d2f)? I'd do it for you, but when you created the PR it seems you must have unticked the box that lets me update this PR.

**Comment by alex-dna** [comment]
@GuySartorelli Sorry about that, I've reset the commits. I'm not sure where the box to let you update the PR is, when I'll make sure I don't untick it inadvertently next time.

**Review by GuySartorelli — APPROVED** [review]
Looks good to me. Thanks for your patience!
